### DISCUSSION TITLE
New Use Case for Stateful CNF

### DISCUSSION THREAD

**Discussion by sishbi** [lfn-cnti/bestpractices#167] [Show and tell]
Hello,
@Smitholi67 and @sishbi have created a new use-case covering a Stateful CNF.
This is captured by the following PR: #165 
We can discuss the use-case either in the PR directly or here.
@Taylor We would like to add this PR / discussion to the agenda for the weekly call.

**Comment by taylor** [comment]
I think it would be good for the glossary to have terms for different types of state.  Eg. ephemeral, cache, user data

**Reply by taylor** [reply]
Watson shared some comments: 

> Different types of people (programmer, network engineer, chaos testing) have different definitions of state and resiliency.
> 
> The main types, for me, are in memory and on disk. What needs to persist and what does not?
> 
> What are the latency requirements?

cc: @sishbi @wavell @denverwilliams

**Reply by sishbi** [reply]
@Taylor When you say 'glossary' do you mean the CNF-WG glossary or the glossary in this use-case?  And just to clarify, you are asking for a description of the different types of state that exist and examples of each? (Such as ephemeral is state that can be lost without consequence, such as a temporary file within a Pod)

**Comment by wvwatson** [comment]
https://github.com/cncf/cnf-wg/tree/main/use-case/0003-UC-stateful-cnf
"The low latency requirements may result in a solution where state is stored in-memory"
"The state that is a maintained by the Stateful CNF must remain ACID complaint, a convergent charging system handles real-time financial transactions"

https://en.wikipedia.org/wiki/In-memory_database#:~:text=ACID%20support,-See%20also%3A%20Atomicity&text=Volatile%20memory%2Dbased%20IMDBs%20can,a%20given%20moment%20in%20time.
"In its simplest form, main memory databases store data on volatile memory devices. These devices lose all stored information when the device loses power or is reset. In this case, **IMDBs can be said to lack support for the "durability" portion** of the ACID (atomicity, consistency, isolation, durability) properties. Volatile memory-based IMDBs can, and often do, support the other three ACID properties of atomicity, consistency and isolation."

Just FYI a lot of old school DBAs don't think in memory DBs are ACID compliant, because of durability.

**Reply by sishbi** [reply]
Thanks @wavell - but if the IMDB is specifically designed to include 'durability' features that allow the memory contents to be recreated - would that be sufficient to be 'ACID' ?
